csm-1.4/CASMINST-6896-gpg-keys/install-hotfix.sh

```diff
@@ -0,0 +1,256 @@
+
+if [ "$running_system_only" -eq 1 ]; then
+  cat >&2 <<EOF
++ Hotfix installed
+${0##*/}: OK
+EOF
+  exit 0
+fi
+
+############################################### CFS ########################################################
+
+workdir="$(mktemp -d)"
+[ -z "${DEBUG:-}" ] && trap 'rm -fr '"${workdir}"'' ERR INT EXIT RETURN || echo "DEBUG was set in environment, $workdir will not be cleaned up."
+
+# Set credentials for the VCS
+PW=$(kubectl -n services get secret vcs-user-credentials -o jsonpath='{.data.vcs_password}' | base64 -d)
+
+# Fetch the latest commit from the specified branch
+CSM_CONFIG_COMMIT=$(git ls-remote "https://crayvcs:${PW}@api-gw-service-nmn.local/vcs/cray/csm-config-management.git" "refs/heads/cray/csm/${CSM_CONFIG_VERSION}" | awk '{print $1}')
+unset PW
+
+[ -z "$CSM_CONFIG_COMMIT" ] && { echo >&2 "Failed to retrieve the latest commit from csm-config branch cray/csm/${CSM_CONFIG_VERSION}. Aborting."; exit 1; }
+
+# Update the kubernetes secret with our new GPG key if the new key isn't present
+NEW_KEY_PATH="${ROOT_DIR}/keys/${GPG_KEY_FILE_NAME}"
+NEW_KEY_SIGNATURE="$(gpg --list-packets "${NEW_KEY_PATH}")"
+NEW_KEY_ENCODED="$(base64 -w 0 "${NEW_KEY_PATH}")"
+mapfile -t EXISTING_K8S_KEYS < <(kubectl -n services get secret hpe-signing-key -o jsonpath='{.data}' | jq -r 'keys[]')
+
+KEY_PRESENT=0
+for key in "${EXISTING_K8S_KEYS[@]}"; do
+  EXISTING_KEY_SIGNATURE="$(kubectl -n services get secret hpe-signing-key -o jsonpath="{.data.${key/./\\.}}" | base64 -d | gpg --list-packets)"
+
+  if [ "${EXISTING_KEY_SIGNATURE}" = "${NEW_KEY_SIGNATURE}" ]; then
+    echo "Key ${key} is already present in the secret. Refusing to add it again."
+    KEY_PRESENT=1
+    break
+  fi
+done
+
+if [ "${KEY_PRESENT}" -eq 0 ]; then
+  echo "Key ${key} is not present, adding it to the secret: hpe-signing-key"
+  kubectl patch secret hpe-signing-key -n services -p="{\"data\":{\"${GPG_KEY_FILE_NAME}\":\"${NEW_KEY_ENCODED}\"}}"
+fi
+
+# Create new manifest.
+cat >"${workdir}/manifest.yaml" <<EOF
+apiVersion: manifests/v1beta1
+metadata:
+  name: casminst-6896-gpg-keys
+spec:
+  sources:
+    charts:
+    - name: nexus
+      type: repo
+      location: https://packages.local/repository/charts
+  charts:
+  - name: csm-config
+    source: nexus
+    version: ${CSM_CONFIG_VERSION}
+    namespace: services
+EOF
+
+# Merge manifest.
+kubectl -n loftsman get secret site-init -o jsonpath='{.data.customizations\.yaml}' | base64 -d >"${workdir}/customizations.yaml"
+manifestgen -c "${workdir}/customizations.yaml" -i "${workdir}/manifest.yaml" -o "${workdir}/deploy-hotfix.yaml"
+
+# Load artifacts into nexus
+"${ROOT_DIR}/lib/setup-nexus.sh"
+
+# Deploy chart.
+loftsman ship --manifest-path "${workdir}/deploy-hotfix.yaml"
+
+# Update sysmgmt chart.
+kubectl -n loftsman get cm loftsman-sysmgmt -o jsonpath='{.data.manifest\.yaml}' >"${workdir}/sysmgmt.yaml"
+yq w -i "${workdir}/sysmgmt.yaml" 'spec.charts.(name==csm-config).version' "${CSM_CONFIG_VERSION}"
```

Comment:
An optional suggestion. Just showing off how to do this in `yq4`.
```suggestion
yq4 eval -i '.spec.charts.(name==csm-config).version = "'"${CSM_CONFIG_VERSION}"'"' "${workdir}/sysmgmt.yaml"
```